app/src/main/java/com/codingblocks/chatter/fragments/RoomsFragment.java

```diff
@@ -288,4 +303,175 @@ protected void onPostExecute(List<RoomsTable> notes) {
             getActivity().finish();
         }
     }
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        inflater.inflate(R.menu.my_options_menu, menu);
+        MenuItem item = menu.findItem(R.id.search);
+        SearchView searchView = (SearchView) item.getActionView();
+        searchView.setOnCloseListener(new SearchView.OnCloseListener() {
+            @Override
+            public boolean onClose() {
+                displayRooms(mRooms);
+                return false;
+            }
+        });
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                return false;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String newText) {
+                mRooms.clear();
+                adapter.notifyDataSetChanged();
+                searchRooms(newText);
+                return true;
+            }
+        });
+        super.onCreateOptionsMenu(menu, inflater);
+    }
+
+    private void searchRooms(String newText) {
+        if (isNetworkAvailable()) {
+            /* Display a toast to inform the user that we are syncing */
+            String accessToken = getActivity()
+                    .getSharedPreferences("UserPreferences", 0)
+                    .getString("accessToken", "");
+            if (accessToken.equals("")) {
+                Intent intent = new Intent(getActivity(), SplashActivity.class);
+                getActivity().startActivity(intent);
+                getActivity().finish();
```

Comment:
This code won't be required as Rooms fragment is embedded inside dashboard fragment which already does this check.